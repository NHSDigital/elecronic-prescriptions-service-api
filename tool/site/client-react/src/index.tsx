import {Bundle} from "fhir/r4"
import * as React from "react"
import {
  createPrescriptionSummaryProps,
  PrescriptionSummary
} from "./components/prescription-summary/prescriptionSummary"
import {PageContainer} from "./components/pageContainer"
import ReactDOM = require("react-dom")

(async function() {
  const baseUrl = "/"
  const urlParams = new URLSearchParams(window.location.search)
  const prescriptionId = urlParams.get("prescription_id")
  const getPrescriptionResponse: Response = await fetch(`${baseUrl}prescription/${prescriptionId}`)
  const bundle: Bundle = await getPrescriptionResponse.json()
  const prescriptionSummaryProps = createPrescriptionSummaryProps(bundle)
  const content = (
    <PageContainer>
      <PrescriptionSummary
        patient={prescriptionSummaryProps.patient}
        practitionerRole={prescriptionSummaryProps.practitionerRole}
      />
    </PageContainer>
  )
  ReactDOM.render(content, document.getElementById("root"))
}())
