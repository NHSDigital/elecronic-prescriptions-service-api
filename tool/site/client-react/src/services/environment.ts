export type Environment = "internal-dev" | "int"

export const internalDev = "internal-dev"
export const int = "int"

export function isDev(environment: Environment): boolean {
  return environment === internalDev
}

export function isInt(environment: Environment): boolean {
  return environment === int
}
