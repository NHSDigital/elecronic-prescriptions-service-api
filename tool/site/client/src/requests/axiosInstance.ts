import axios from "axios"

export const axiosInstance = axios.create({
  validateStatus: () => true
})

// todo: refresh token if required
axiosInstance.interceptors.request.use(function (config) {
  return config
}, function (error) {
  return Promise.reject(error)
})
