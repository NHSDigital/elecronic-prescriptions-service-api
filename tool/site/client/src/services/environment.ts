export type Environment = "internal-dev" | "internal-dev-sandbox" | "internal-qa" | "int" | "sandbox"

export const internalDev = "internal-dev"
export const internalDevSandbox = "internal-dev-sandbox"
export const internalQa = "internal-qa"
export const int = "int"
export const sandbox = "sandbox"

export function isDev(environment: Environment): boolean {
  return environment === internalDev || environment === internalDevSandbox
}

export function isQa(environment: Environment): boolean {
  return environment === internalQa
}

export function isInt(environment: Environment): boolean {
  return environment === int
}

export function isSandbox(environment: Environment): boolean {
  return environment === sandbox
}
