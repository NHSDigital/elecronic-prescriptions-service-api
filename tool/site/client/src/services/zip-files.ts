import JSZip from "jszip"
import FileSaver from "file-saver"

interface ZipFileItem {
  fileName: string
  data: string
}

export const zip = async (zipFileName: string, itemsToBeZipped: Array<ZipFileItem>): Promise<void> => {
  const zip = JSZip()

  itemsToBeZipped.forEach(item => {
    zip.file(item.fileName, item.data)
  })

  zip.generateAsync({type: "blob"}).then(zipFile => {
    return FileSaver.saveAs(zipFile, `${zipFileName}.zip`)
  })
}
