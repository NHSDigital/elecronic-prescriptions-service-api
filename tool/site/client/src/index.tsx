import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
