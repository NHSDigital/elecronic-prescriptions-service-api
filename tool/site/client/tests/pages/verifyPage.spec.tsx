import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import {axiosInstance} from "../../src/requests/axiosInstance"
import {internalDev} from "../../src/services/environment"
import VerifyPage from "../../src/pages/verifyPage"
import {readMessage} from "../messages/messages"

const baseUrl = "baseUrl/"
const prescriptionId = "7A9089-A83008-56A03J"
const context: AppContextValue = {baseUrl, environment: internalDev}

const releaseUrl = `${baseUrl}dispense/release/${prescriptionId}`
const verifyUrl = `${baseUrl}dispense/verify`

const releaseResponse = readMessage("releaseResponse.json")

beforeEach(() => moxios.install(axiosInstance))

afterEach(() => moxios.uninstall(axiosInstance))

test("Displays verify form when no prescription id is provided", async () => {
  const container = await renderPage()

  expect(screen.getByText("Verify prescription(s)")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays verify result when prescription id is provided", async () => {
  moxios.stubRequest(releaseUrl, {
    status: 200,
    response: releaseResponse
  })
  moxios.stubRequest(verifyUrl, {
    status: 200,
    response: {
      results: [{
        name: "0",
        success: true
      }],
      success: true,
      request: "JSON Request",
      response: "JSON Response"
    }
  })

  const container = await renderPage(prescriptionId)

  expect(screen.getByText("Verify Result")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage(prescriptionId?: string) {
  const {container} = renderWithContext(<VerifyPage prescriptionId={prescriptionId}/>, context)
  if (prescriptionId) {
    await waitFor(() => screen.getByText("Verify Result"))
  } else {
    await waitFor(() => screen.getByText("Verify prescription(s)"))
  }
  return container
}
