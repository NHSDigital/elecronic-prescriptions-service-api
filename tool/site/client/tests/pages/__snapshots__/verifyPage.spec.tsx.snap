// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays verify form when no prescription id is provided 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Verify prescription(s)</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <div class=\\"nhsuk-form-group\\"><textarea class=\\"nhsuk-textarea\\" name=\\"verifyRequest\\" id=\\"verifyRequest\\" rows=\\"20\\"></textarea></div>
    </fieldset>
  </div>
  <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Verify</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>
</form>"
`;

exports[`Displays verify result when prescription id is provided 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Verify Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<div class=\\"nhsuk-table-responsive\\">
  <table class=\\"nhsuk-table\\">
    <thead class=\\"nhsuk-table__head\\">
      <tr class=\\"nhsuk-table__row\\">
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Signature Name</th>
        <th class=\\"nhsuk-table__header\\" scope=\\"col\\">Success</th>
      </tr>
    </thead>
    <tbody class=\\"nhsuk-table__body\\">
      <tr class=\\"nhsuk-table__row\\">
        <td class=\\"nhsuk-table__cell\\">0</td>
        <td class=\\"nhsuk-table__cell\\"><svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
            <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
          </svg></td>
      </tr>
    </tbody>
  </table>
</div>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Request%22\\" download=\\"message\\">Download</a></div>
      <div class=\\"react-json-view\\" style=\\"font-family: monospace; cursor: default; background-color: rgb(255, 255, 255); position: relative; margin-top: 10px;\\">
        <div class=\\"pretty-json-container object-container\\">
          <div class=\\"object-content\\">
            <div class=\\"object-key-val\\"><span><span style=\\"display: inline-block; cursor: pointer;\\"><div class=\\"icon-container\\" style=\\"display: inline-block; width: 17px;\\"><span class=\\"expanded-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); height: 1em; width: 1em; padding-left: 2px;\\" viewBox=\\"0 0 15 15\\" fill=\\"currentColor\\"><path d=\\"M0 5l6 6 6-6z\\"></path></svg></span></div><span style=\\"display: inline-block; color: rgb(0, 94, 184); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;\\"><span class=\\"object-key\\"><span style=\\"vertical-align: top;\\">\\"</span><span>ERROR</span><span style=\\"vertical-align: top;\\">\\"</span></span><span style=\\"display: inline-block; margin: 0px 5px; color: rgb(0, 94, 184); vertical-align: top;\\">:</span></span><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184);\\">{</span></span>
            <div style=\\"display: inline-block; padding: 0px 0px 0px 10px;\\" class=\\"object-meta-data\\"><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span></div></span>
            <div class=\\"pushed-content object-container\\">
              <div class=\\"object-content\\" style=\\"margin-left: 6px;\\">
                <div style=\\"border-left: 1px solid #000000; padding: 3px 5px 3px 20px;\\" class=\\"variable-row\\"><span><span style=\\"display: inline-block; color: rgb(0, 94, 184); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;\\" class=\\"object-key\\"><span style=\\"vertical-align: top;\\">\\"</span><span style=\\"display: inline-block;\\">message</span><span style=\\"vertical-align: top;\\">\\"</span></span><span style=\\"display: inline-block; margin: 0px 5px; color: rgb(0, 94, 184); vertical-align: top;\\">:</span></span>
                  <div class=\\"variable-value\\" style=\\"display: inline-block; padding-right: 6px; position: relative; cursor: default;\\">
                    <div style=\\"display: inline-block; color: rgb(0, 0, 0);\\"><span class=\\"string-value\\" style=\\"cursor: default;\\">\\"src property must be a valid json object\\"</span></div>
                  </div><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span>
                </div>
              </div>
            </div><span class=\\"brace-row\\"><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184); padding-left: 3px;\\">}</span></span>
          </div>
        </div>
      </div>
    </div>
</div>
</details>
<details class=\\"nhsuk-details nhsuk-expander\\">
  <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (FHIR)</span></summary>
  <div class=\\"nhsuk-details__text\\">
    <div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" aria-disabled=\\"false\\" type=\\"submit\\">Copy</button><a class=\\"nhsuk-button sc-jRQBWg fkXsvN\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Response%22\\" download=\\"message\\">Download</a></div>
    <div class=\\"react-json-view\\" style=\\"font-family: monospace; cursor: default; background-color: rgb(255, 255, 255); position: relative; margin-top: 10px;\\">
      <div class=\\"pretty-json-container object-container\\">
        <div class=\\"object-content\\">
          <div class=\\"object-key-val\\"><span><span style=\\"display: inline-block; cursor: pointer;\\"><div class=\\"icon-container\\" style=\\"display: inline-block; width: 17px;\\"><span class=\\"expanded-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); height: 1em; width: 1em; padding-left: 2px;\\" viewBox=\\"0 0 15 15\\" fill=\\"currentColor\\"><path d=\\"M0 5l6 6 6-6z\\"></path></svg></span></div><span style=\\"display: inline-block; color: rgb(0, 94, 184); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;\\"><span class=\\"object-key\\"><span style=\\"vertical-align: top;\\">\\"</span><span>ERROR</span><span style=\\"vertical-align: top;\\">\\"</span></span><span style=\\"display: inline-block; margin: 0px 5px; color: rgb(0, 94, 184); vertical-align: top;\\">:</span></span><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184);\\">{</span></span>
          <div style=\\"display: inline-block; padding: 0px 0px 0px 10px;\\" class=\\"object-meta-data\\"><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span></div></span>
          <div class=\\"pushed-content object-container\\">
            <div class=\\"object-content\\" style=\\"margin-left: 6px;\\">
              <div style=\\"border-left: 1px solid #000000; padding: 3px 5px 3px 20px;\\" class=\\"variable-row\\"><span><span style=\\"display: inline-block; color: rgb(0, 94, 184); letter-spacing: 0.5px; font-style: none; vertical-align: top; opacity: 0.85;\\" class=\\"object-key\\"><span style=\\"vertical-align: top;\\">\\"</span><span style=\\"display: inline-block;\\">message</span><span style=\\"vertical-align: top;\\">\\"</span></span><span style=\\"display: inline-block; margin: 0px 5px; color: rgb(0, 94, 184); vertical-align: top;\\">:</span></span>
                <div class=\\"variable-value\\" style=\\"display: inline-block; padding-right: 6px; position: relative; cursor: default;\\">
                  <div style=\\"display: inline-block; color: rgb(0, 0, 0);\\"><span class=\\"string-value\\" style=\\"cursor: default;\\">\\"src property must be a valid json object\\"</span></div>
                </div><span class=\\"copy-to-clipboard-container\\" title=\\"Copy to clipboard\\" style=\\"vertical-align: top; display: none;\\"><span style=\\"cursor: pointer; display: inline;\\"><span class=\\"copy-icon\\"><svg style=\\"vertical-align: top; color: rgb(0, 0, 0); font-size: 15px; margin-right: 3px; height: 1em; width: 1em;\\" viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" preserveAspectRatio=\\"xMidYMid meet\\"><g><path d=\\"m30 35h-25v-22.5h25v7.5h2.5v-12.5c0-1.4-1.1-2.5-2.5-2.5h-7.5c0-2.8-2.2-5-5-5s-5 2.2-5 5h-7.5c-1.4 0-2.5 1.1-2.5 2.5v27.5c0 1.4 1.1 2.5 2.5 2.5h25c1.4 0 2.5-1.1 2.5-2.5v-5h-2.5v5z m-20-27.5h2.5s2.5-1.1 2.5-2.5 1.1-2.5 2.5-2.5 2.5 1.1 2.5 2.5 1.3 2.5 2.5 2.5h2.5s2.5 1.1 2.5 2.5h-20c0-1.5 1.1-2.5 2.5-2.5z m-2.5 20h5v-2.5h-5v2.5z m17.5-5v-5l-10 7.5 10 7.5v-5h12.5v-5h-12.5z m-17.5 10h7.5v-2.5h-7.5v2.5z m12.5-17.5h-12.5v2.5h12.5v-2.5z m-7.5 5h-5v2.5h5v-2.5z\\"></path></g></svg></span></span></span>
              </div>
            </div>
          </div><span class=\\"brace-row\\"><span style=\\"display: inline-block; cursor: pointer; font-weight: bold; color: rgb(0, 94, 184); padding-left: 3px;\\">}</span></span>
        </div>
      </div>
    </div>
  </div>
  </div>
</details>
</div>
<div class=\\"sc-hKwDye gqtOcv\\"><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"button\\">Back</button></div>"
`;
